### ISSUE TITLE
mupen64plus randomly fails to overwrite and load savestates

### ISSUE THREAD

**Issue by Redyn** [mupen64plus/mupen64plus-core#695]
Sometimes when I try to overwrite a preexisting savestate, mupen64plus fails to do it, yet there's no error warning about it. Then, if I try loading said savestate, the emulator hangs, forcing me to close it with the task manager.

This basically renders games unplayable when using savestates, as I have no way of knowing when the issue will strike.

I'm on Windows 10. This problem never occurred when I used Windows 7, although it may be because I used a previous version on it. I tried setting the compatibility mode to Windows 7 and the issue persists. This isn't a file access permission issue, as I'm using an admin account that has full control of the Savestates directory, and mupen64lpus can write new savestates just fine - the problem seems to occur only when trying to overwrite a previous savestate. And even then, it occurs seemingly at random (sometimes I can successfully overwrite them, other times I can't).

**Comment by Redyn** [comment]
Welp, looks like it's back to Project64 for me.

**Comment by richard42** [comment]
If you encounter this bug and kill the emulator with the task manager, and restart it, can you then load the savestate?

**Comment by Redyn** [comment]
Yes, but only the original savestate. The new savestate that attempted to overwrite it didn't get saved.

**Comment by richard42** [comment]
I looked through the savestate code yesterday. It used to be quite simple but has become more complicated in the last few years, and may be getting into a deadlock state for some reason. 2 more questions:
Are you using the keyboard-shortcut savestate slots (0-9) or saving the state with a GUI application?
and, what game(s) are you playing when this happens?

**Comment by Redyn** [comment]
Yup, keyboard shortcuts.

Banjo-Tooie. The same game presented no such issue when I played it in Win 7.

**Comment by richard42** [comment]
Are you running the 2.5.9 release build? I think this problem may have been fixed since that release. I tested this tonight on my Linux computer, with Banjo Tooie. I probably saved and reloaded 30 times, and it did lock up one time when I saved the state. Then I discovered that I was actually running an old version (I forgot to do an update before I built), and the code which had caused it to lock up was removed during some interrupt refactoring work that was merged in February.

Can you test with the latest code?  You can download 32 and 64 bit builds for Windows from Sven's site here:

https://bitbucket.org/ecsv/mupen64plus-mxe-daily/src/master/
